Proposal: WACI-Presentation-Exchange Work Item (#5)
decentralized-identity/claims-credentials: work_items/WACI_presentation_exchange.md

# WACI Presentation Exchange

## Work Item Owners
- Orie Steele (@OR13)
- Brent Zundel (@brentzundel)

## Outcome
**When your work item is successful, then what will have changed in the world? What
will we be able to do that wasn't possible do before or vice versa?**

We will have a standard protocol for requesting and submitting a presentation of
a variety of credentials formats. 

## Deliverables
- **Does your work item have any deliverables? These include code projects,
  standards-track specifications, reports, registries, summaries, blog posts,
  and anything else that is instantiable and publishable in written form to a
  broader audience.**

> We are writing a standards-track specification.

Reviewer: Should we specifically mention test-vectors or interop profiles here or are these covered via https://github.com/decentralized-identity/org/blob/master/Org%20documents/Membership%20agreements/Project%20Charter%20Addendum.md

Reviewer 2: Proposed: https://github.com/decentralized-identity/claims-credentials/pull/5/files#r620548576

> We are writing a standards-track specification.

Reviewer 2: ```suggestion
We are writing a standards-track specification, which will include a complete set of test vectors covering all messages exchanged by the protocol.
```

> 

Reviewer 2: We will provide example payloads (test vectors) associated with the spec.

- **What's the expected timeline?**

> We will release v0.1 for developer use by .

Reviewer 2: ```suggestion
We will release v0.1 for developer use by June 1 2021
```

> We will release v0.1 for developer use by .

Reviewer: Can we move in the preliminary timeline from IIW?

Reviewer 2: Pasting from the notes:

```
May 1: GHP Drafting Groups First Drafts Due - 

            June 1 GHP Interoperability Blueprint Recommendations Spec complete
30 day vision
90 day vision  <- would be ideal to have something that can be tested against for this timeframe. 
180 day vision

            July 1 Test Complete

            August 1 - 10+ Implementations / Vendors passing VP Exchange Interop Tests.
	
 	October 1 - Cross Wallet Interop Exchange Tests.
```

# WACI Presentation Exchange

## Work Item Owners
- Orie Steele (@OR13)

> - Brent Zundel (@brentzundel)

Author: ```suggestion
- Brent Zundel (@brentzundel)
- Jace Hensley (@JaceHensley)

```

## Outcome
**When your work item is successful, then what will have changed in the world? What
will we be able to do that wasn't possible do before or vice versa?**

We will have a standard protocol for requesting and submitting a presentation of
a variety of credentials formats. 

## Deliverables
- **Does your work item have any deliverables? These include code projects,
  standards-track specifications, reports, registries, summaries, blog posts,
  and anything else that is instantiable and publishable in written form to a
  broader audience.**

We are writing a standards-track specification, which will include a complete
set of test vectors covering all messages exchanged by the protocol.

- **What's the expected timeline?**

> We will release v0.1 for developer use by June 1 2021

Author: ```suggestion
We will release v0.1 for developer use by June 1 2021, then continue working toward v1.0 with timeline TBD.
```

- **Who's the audience of your deliverables?**

We are writing for developers who need to ask for and exchange claims that are

> verifiable, such as Verifiable Credentials.

Author: ```suggestion
verifiable, such as Verifiable Credentials expressed in a Verifiable Presentation.
```

- **What are related resources or standards from DIF or adjacent
  organizations?**

> JSON-LD Signatures, JWT, DIF Presentation-Exchange, DIDComm, and WACI.

Author: ```suggestion
JSON-LD Signatures, JWT, DIF Presentation-Exchange and Credential Manifest, DIDComm,
CCG VC HTTP API, GNAP, DPoP, OIDF, SIOP, and WACI.
```

## Meetings
- **Are there regular meetings? If so, when?**

Yes, Mondays at 1 PM PT/4 PM ET at [this Zoom link](https://us02web.zoom.us/j/88301532931?pwd=d0pISWZWNFozU3RkTmJxME0rd1FBZz09).

- **What are the goals of the meeting?**

To further work on the specification.

- **Do you have specific discussion topics in mind, or is it meant to move a
  deliverable forward?**

To move our deliverable forward, we will review the outstanding GitHub issues,
using [this link](https://github.com/decentralized-identity/waci-presentation-exchange/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-asc).

When we have specific goals such as a release target (e.g., `v0.1`) or items
labeled "High Priority", then we may add additional filters for going through
issues.

- **Where are the notes and recordings published, if any?**

The recordings are published in the #general channel on the DIF Slack. There
are no notes, however automated transcription is a possibility and actions are
mainly taken in GitHub issues.

- **Who might be interested in attending?**

> Anyone interested in a protocol for asking for and receiving presentations.

Author: ```suggestion
Anyone interested in a protocol for using credentials.
```
